### PR TITLE
Add docs for aws_iam_role + clean up examples in IAM docs

### PR DESCRIPTION
**Pull request by radeksimko**
Closing #1882 + little bit of cleanup.

### CODE REVIEW

File: website/source/docs/providers/aws/r/iam_role_policy.html.markdown [hunk 1]
```diff
@@ -13,6 +13,23 @@ Provides an IAM role policy.
...
+  "Statement": [
+    {
+      "Action": [
+        "ec2:Describe*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
```
Comment (catsby) [inline website/source/docs/providers/aws/r/iam_role_policy.html.markdown]: missing `EOF`?

Comment (radeksimko) [inline website/source/docs/providers/aws/r/iam_role_policy.html.markdown]: Ah, good spot, fix coming :steam_locomotive:

**Comment by radeksimko** [comment]
@catsby Missing EOF added

**Comment by catsby** [comment]
Looks good from here :+1:

**Pull request merged**